### DISCUSSION TITLE
Need help: how to manually build whl

### DISCUSSION THREAD

**Discussion by hkvision** [apache/incubator-mxnet#17587] [Q&A] [answered]
Hi,

Are there any instructions to build the whl for mxnet that is equivalent to the whl that you release on pypi?

I have compiled the project using mkl with libmxnet.so under incubator-mxnet/lib, then I run `python setup.py bdist_wheel` under `incubator-mxnet/python` and get a whl under `dist`, if I pip install this whl, I am encounting the following issues when import mxnet:
- Cannot find libmxnet.so, I am using conda, and seems the file is under `/root/anaconda3/envs/py36/mxnet/libmxnet.so` instead of in `/root/anaconda3/envs/py36/lib/python3.6/site-packages/mxnet/` and cannot be found by code.
- Wondering will the so files for libopencv included?
- dmlc_tracker package is not installed. However, if I install from released pypi, it is installed automatically. Related issue: https://discuss.mxnet.io/t/cant-load-dmlc-tracker-package/1282/3

Thanks so much in advance!

**Comment by stu1130** [comment]
[@mxnet-label-bot add Question]

**Comment by stu1130** [comment]
@ChaiBapchya could you take a look at this question?

**Comment by leezu** [comment]
@hkvision this is a known issue. Workaround: https://github.com/apache/incubator-mxnet/issues/16933#issuecomment-561461776

@ChaiBapchya if you could help fix it, that would be great.

**Comment by ChaiBapchya** [comment]
Working on it.

**Comment by hkvision** [comment]
Hi @leezu Thanks for supporting. I saw the workaround for libmxnet.so in the comment. I will follow https://github.com/apache/incubator-mxnet/tree/master/tools/staticbuild and the way you mention in the comment. 
Wondering will the remaining two points I mentioned in the issue description (dmlc, opencv) be resolved in this case?
Thanks so much.

**Comment by leezu** [comment]
OpenCV will be linked statically.

@szha please advise about dmlc_tracker. You mentioned on Jun '18 2018

> pip package will include dmlc_tracker starting from tonight’s nightly build.

**Comment by szha** [comment]
@leezu it's copied into the wheel from `3rdparty/dmlc-core/tracker/dmlc_tracker`.

**Comment by szha** [comment, marked as answer]
The staticbuild scripts for building custom whl are now available in the MXNet code base https://github.com/apache/incubator-mxnet/tree/master/tools/staticbuild
